server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -0,0 +1,363 @@
+                    getStdOutMsg(input, result.orElse(null)));
+        } else {
+            emitMessage(this.producer, String.format("fasten.%s.err",
+                    plugin.getClass().getSimpleName()),
+                    getStdErrMsg(input));
+        }
+    }
+
+    /**
+     * Send message to Kafka topic.
+     *
+     * @param producer Kafka producer
+     * @param topic    topic to send to
+     * @param msg      message
+     */
+    private void emitMessage(KafkaProducer<String, String> producer, String topic, String msg) {
+        ProducerRecord<String, String> errorRecord = new ProducerRecord<>(topic, msg);
+
+        producer.send(errorRecord, (recordMetadata, e) -> {
+            if (recordMetadata != null) {
+                logger.debug("Sent: {} to {}", msg, this.serverLogTopic);
+            } else {
+                e.printStackTrace();
+            }
+        });
+
+        producer.flush();
+    }
+
+    /**
+     * Create a message that will be send to STDOUT of a given plugin.
+     *
+     * @param input   consumed record
+     * @param payload output of the plugin
+     * @return stdout message
+     */
+    private String getStdOutMsg(String input, String payload) {
+        JSONObject stdoutMsg = new JSONObject();
+        stdoutMsg.put("created_at", System.currentTimeMillis());
+        stdoutMsg.put("plugin_name", plugin.getClass().getSimpleName());
+        stdoutMsg.put("plugin_version", plugin.version());
+
+        stdoutMsg.put("input", StringUtils.isNotEmpty(input) ? new JSONObject(input) : "");
+        stdoutMsg.put("payload", StringUtils.isNotEmpty(payload) ? new JSONObject(payload) : "");
+
+        return stdoutMsg.toString();
+    }
+
+    /**
+     * Create a message that will be send to STDERR of a given plugin.
+     *
+     * @param input consumed record
+     * @return stderr message
+     */
+    private String getStdErrMsg(String input) {
+        JSONObject stderrMsg = new JSONObject();
+        stderrMsg.put("created_at", System.currentTimeMillis() / 1000L);
+        stderrMsg.put("plugin_name", plugin.getClass().getSimpleName());
+        stderrMsg.put("plugin_version", plugin.version());
+
+        stderrMsg.put("input", !Strings.isNullOrEmpty(input) ? new JSONObject(input) : "");
+
+        Throwable pluginError = plugin.getPluginError();
+        JSONObject error = new JSONObject();
+        error.put("error", pluginError.getClass().getSimpleName());
+        error.put("msg", pluginError.getMessage());
+        error.put("stacktrace", pluginError.getStackTrace());
+
+        stderrMsg.put("err", error);
+
+        return stderrMsg.toString();
+    }
+
+    /**
+     * This is a synchronous commits and will block until either the commit succeeds
+     * or an unrecoverable error is encountered.
+     */
+    private void doCommitSync() {
+        try {
+            connection.commitSync();
+            logger.debug("Committed the processed record...");
```

Comment:
What is the point of this log message? What record? And why the ellipses at the end?

Reply:
Removed this log